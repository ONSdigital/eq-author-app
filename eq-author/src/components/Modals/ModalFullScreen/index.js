import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { colors } from "constants/theme";
import Modal, { CloseButton } from "components/Modals/Modal";
import ScrollPane from "components/ScrollPane";

const topOffset = "4em"; //Header height

const StyledModal = styled(Modal)`
  .Overlay {
    background-color: transparent;
  }
  .Modal {
    width: 100%;
    top: ${topOffset};
    height: calc(100% - ${topOffset});
    background: ${colors.lighterGrey};
    padding: 0;
  }
`;

const ModalFullScreen = props => {
  const { children, onClose, isOpen, ...otherProps } = props;
  return (
    <StyledModal
      isOpen={isOpen}
      onClose={onClose}
      hasCloseButton={false}
      {...otherProps}
    >
      <ScrollPane>
        <CloseButton onClick={onClose}>&times;</CloseButton>
        {children}
      </ScrollPane>
    </StyledModal>
  );
};

ModalFullScreen.propTypes = {
  children: PropTypes.node.isRequired,
  onClose: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired
};

export default ModalFullScreen;
