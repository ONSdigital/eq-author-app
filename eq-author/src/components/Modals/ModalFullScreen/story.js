import React from "react";
import { MemoryRouter } from "react-router";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import styled from "styled-components";

import { colors } from "constants/theme";
import ModalFullScreen from "./";
import DialogHeader from "components/Dialog/DialogHeader";
import { Message, Heading } from "components/Dialog/DialogMessage";
import Button from "components/Buttons/Button";

const CenteredHeading = styled(Heading)`
  padding: 2em;
  text-align: center;
  color: ${colors.textLight};
`;

const MainContainer = styled.div`
  padding: 2em 3em;
  min-height: 30em;
  display: flex;
  flex-direction: column;
`;

const Content = styled.div`
  flex: 1 1 auto;
`;

const Buttons = styled.div`
  display: flex;
  justify-content: flex-end;
`;

const handleOnClose = action("Close");

storiesOf("ModalFullScreen", module)
  .addDecorator(story => (
    <MemoryRouter initialEntries={["/"]}>{story()}</MemoryRouter>
  ))
  .add("With scrollable content", () => (
    <div>
      <ModalFullScreen onClose={handleOnClose} isOpen>
        <DialogHeader>
          <Message>
            <CenteredHeading>Ridiculus Tellus Sit</CenteredHeading>
          </Message>
        </DialogHeader>
        <MainContainer>
          <Content>
            <p>
              Lorem ipsum dolor sit amet, magna pede, ac ac id condimentum,
              adipiscing lacinia arcu porttitor, tortor condimentum, ultrices
              odio orci ipsum sodales. Viverra odio maecenas volutpat sapien
              sed, semper erat pulvinar quis vitae fringilla consequatur. Ac
              pharetra lacus lectus wisi scelerisque, quisque gravida etiam
              lorem felis aenean condimentum, neque mauris. Aliquam a vel justo
              phasellus, vivamus posuere suspendisse nunc, placerat dui ipsum,
              molestie vitae hymenaeos pulvinar integer id tellus, convallis
              pellentesque qui luctus ut. Est pellentesque sit porta duis donec,
              amet orci a sociosqu primis, ac vehicula nec tortor, vitae ipsum a
              habitasse. Ut ipsum non at. Metus nec donec sed, in leo. Ante
              tellus per amet diam egestas, cum non nibh eu, justo libero mi
              libero scelerisque malesuada, quis sed nisi, massa a sagittis
              orci. Cursus dui in a elementum, lectus massa mauris proin.
              Curabitur habitant laoreet purus aenean aenean, nonummy blandit
              mauris habitant, donec elit. Rutrum quis nulla aliquam, a egestas
              ut, amet congue enim cubilia est sed. A sit voluptatem litora
              eget. Nullam lorem lorem. Vel in lacus sed hac nec, sodales
              natoque, lectus pharetra, quis dis mi purus fusce et commodo. Elit
              morbi urna luctus euismod integer vulputate. Id elementum vitae
              nisl, volutpat elit tellus consectetuer, varius tincidunt vivamus
              tortor, urna ipsum lacus bibendum molestie, tempor non sed felis
              rutrum.
            </p>
            <p>
              Lorem ipsum dolor sit amet, magna pede, ac ac id condimentum,
              adipiscing lacinia arcu porttitor, tortor condimentum, ultrices
              odio orci ipsum sodales. Viverra odio maecenas volutpat sapien
              sed, semper erat pulvinar quis vitae fringilla consequatur. Ac
              pharetra lacus lectus wisi scelerisque, quisque gravida etiam
              lorem felis aenean condimentum, neque mauris. Aliquam a vel justo
              phasellus, vivamus posuere suspendisse nunc, placerat dui ipsum,
              molestie vitae hymenaeos pulvinar integer id tellus, convallis
              pellentesque qui luctus ut. Est pellentesque sit porta duis donec,
              amet orci a sociosqu primis, ac vehicula nec tortor, vitae ipsum a
              habitasse. Ut ipsum non at. Metus nec donec sed, in leo. Ante
              tellus per amet diam egestas, cum non nibh eu, justo libero mi
              libero scelerisque malesuada, quis sed nisi, massa a sagittis
              orci. Cursus dui in a elementum, lectus massa mauris proin.
              Curabitur habitant laoreet purus aenean aenean, nonummy blandit
              mauris habitant, donec elit. Rutrum quis nulla aliquam, a egestas
              ut, amet congue enim cubilia est sed. A sit voluptatem litora
              eget. Nullam lorem lorem. Vel in lacus sed hac nec, sodales
              natoque, lectus pharetra, quis dis mi purus fusce et commodo. Elit
              morbi urna luctus euismod integer vulputate. Id elementum vitae
              nisl, volutpat elit tellus consectetuer, varius tincidunt vivamus
              tortor, urna ipsum lacus bibendum molestie, tempor non sed felis
              rutrum.
            </p>
            <p>
              Lorem ipsum dolor sit amet, magna pede, ac ac id condimentum,
              adipiscing lacinia arcu porttitor, tortor condimentum, ultrices
              odio orci ipsum sodales. Viverra odio maecenas volutpat sapien
              sed, semper erat pulvinar quis vitae fringilla consequatur. Ac
              pharetra lacus lectus wisi scelerisque, quisque gravida etiam
              lorem felis aenean condimentum, neque mauris. Aliquam a vel justo
              phasellus, vivamus posuere suspendisse nunc, placerat dui ipsum,
              molestie vitae hymenaeos pulvinar integer id tellus, convallis
              pellentesque qui luctus ut. Est pellentesque sit porta duis donec,
              amet orci a sociosqu primis, ac vehicula nec tortor, vitae ipsum a
              habitasse. Ut ipsum non at. Metus nec donec sed, in leo. Ante
              tellus per amet diam egestas, cum non nibh eu, justo libero mi
              libero scelerisque malesuada, quis sed nisi, massa a sagittis
              orci. Cursus dui in a elementum, lectus massa mauris proin.
              Curabitur habitant laoreet purus aenean aenean, nonummy blandit
              mauris habitant, donec elit. Rutrum quis nulla aliquam, a egestas
              ut, amet congue enim cubilia est sed. A sit voluptatem litora
              eget. Nullam lorem lorem. Vel in lacus sed hac nec, sodales
              natoque, lectus pharetra, quis dis mi purus fusce et commodo. Elit
              morbi urna luctus euismod integer vulputate. Id elementum vitae
              nisl, volutpat elit tellus consectetuer, varius tincidunt vivamus
              tortor, urna ipsum lacus bibendum molestie, tempor non sed felis
              rutrum.
            </p>
            <p>
              Lorem ipsum dolor sit amet, magna pede, ac ac id condimentum,
              adipiscing lacinia arcu porttitor, tortor condimentum, ultrices
              odio orci ipsum sodales. Viverra odio maecenas volutpat sapien
              sed, semper erat pulvinar quis vitae fringilla consequatur. Ac
              pharetra lacus lectus wisi scelerisque, quisque gravida etiam
              lorem felis aenean condimentum, neque mauris. Aliquam a vel justo
              phasellus, vivamus posuere suspendisse nunc, placerat dui ipsum,
              molestie vitae hymenaeos pulvinar integer id tellus, convallis
              pellentesque qui luctus ut. Est pellentesque sit porta duis donec,
              amet orci a sociosqu primis, ac vehicula nec tortor, vitae ipsum a
              habitasse. Ut ipsum non at. Metus nec donec sed, in leo. Ante
              tellus per amet diam egestas, cum non nibh eu, justo libero mi
              libero scelerisque malesuada, quis sed nisi, massa a sagittis
              orci. Cursus dui in a elementum, lectus massa mauris proin.
              Curabitur habitant laoreet purus aenean aenean, nonummy blandit
              mauris habitant, donec elit. Rutrum quis nulla aliquam, a egestas
              ut, amet congue enim cubilia est sed. A sit voluptatem litora
              eget. Nullam lorem lorem. Vel in lacus sed hac nec, sodales
              natoque, lectus pharetra, quis dis mi purus fusce et commodo. Elit
              morbi urna luctus euismod integer vulputate. Id elementum vitae
              nisl, volutpat elit tellus consectetuer, varius tincidunt vivamus
              tortor, urna ipsum lacus bibendum molestie, tempor non sed felis
              rutrum.
            </p>
            <p>
              Lorem ipsum dolor sit amet, magna pede, ac ac id condimentum,
              adipiscing lacinia arcu porttitor, tortor condimentum, ultrices
              odio orci ipsum sodales. Viverra odio maecenas volutpat sapien
              sed, semper erat pulvinar quis vitae fringilla consequatur. Ac
              pharetra lacus lectus wisi scelerisque, quisque gravida etiam
              lorem felis aenean condimentum, neque mauris. Aliquam a vel justo
              phasellus, vivamus posuere suspendisse nunc, placerat dui ipsum,
              molestie vitae hymenaeos pulvinar integer id tellus, convallis
              pellentesque qui luctus ut. Est pellentesque sit porta duis donec,
              amet orci a sociosqu primis, ac vehicula nec tortor, vitae ipsum a
              habitasse. Ut ipsum non at. Metus nec donec sed, in leo. Ante
              tellus per amet diam egestas, cum non nibh eu, justo libero mi
              libero scelerisque malesuada, quis sed nisi, massa a sagittis
              orci. Cursus dui in a elementum, lectus massa mauris proin.
              Curabitur habitant laoreet purus aenean aenean, nonummy blandit
              mauris habitant, donec elit. Rutrum quis nulla aliquam, a egestas
              ut, amet congue enim cubilia est sed. A sit voluptatem litora
              eget. Nullam lorem lorem. Vel in lacus sed hac nec, sodales
              natoque, lectus pharetra, quis dis mi purus fusce et commodo. Elit
              morbi urna luctus euismod integer vulputate. Id elementum vitae
              nisl, volutpat elit tellus consectetuer, varius tincidunt vivamus
              tortor, urna ipsum lacus bibendum molestie, tempor non sed felis
              rutrum.
            </p>
          </Content>
          <Buttons>
            <Button
              onClick={handleOnClose}
              variant="primary"
              data-test="btn-done"
            >
              Done
            </Button>
          </Buttons>
        </MainContainer>
      </ModalFullScreen>
    </div>
  ));
