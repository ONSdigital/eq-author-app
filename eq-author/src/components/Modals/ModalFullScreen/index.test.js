import React from "react";
import { shallow } from "enzyme";
import ModalFullScreen from "./";

const createWrapper = (props, render = shallow) => {
  return render(<ModalFullScreen {...props} />);
};

describe("ModalFullScreen", () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      onClose: jest.fn(),
      isOpen: false,
      children: <div>Modal content</div>
    };

    wrapper = createWrapper(props);
  });

  it("should render when closed", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render when open", () => {
    wrapper = createWrapper({
      ...props,
      isOpen: true
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("should call close handler when closed", () => {
    wrapper.simulate("close");
    expect(props.onClose).toHaveBeenCalled();
  });
});
