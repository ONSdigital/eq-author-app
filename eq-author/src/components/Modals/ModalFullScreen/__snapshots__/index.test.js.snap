// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalFullScreen should render when closed 1`] = `
<ModalFullScreen__StyledModal
  hasCloseButton={false}
  isOpen={false}
  onClose={[MockFunction]}
>
  <ScrollPane
    permanentScrollBar={false}
  >
    <Modal__CloseButton
      onClick={[MockFunction]}
    >
      ×
    </Modal__CloseButton>
    <div>
      Modal content
    </div>
  </ScrollPane>
</ModalFullScreen__StyledModal>
`;

exports[`ModalFullScreen should render when open 1`] = `
<ModalFullScreen__StyledModal
  hasCloseButton={false}
  isOpen={true}
  onClose={[MockFunction]}
>
  <ScrollPane
    permanentScrollBar={false}
  >
    <Modal__CloseButton
      onClick={[MockFunction]}
    >
      ×
    </Modal__CloseButton>
    <div>
      Modal content
    </div>
  </ScrollPane>
</ModalFullScreen__StyledModal>
`;
